dyo/configuration.go

```diff
@@ -0,0 +1,282 @@
+				{"RECAPTCHA_SITE_KEY", ""},
+			},
+			Volumes: []Volume{{"crux-certs", "/app/certs", true}},
+			Ports:   []Port{{3000, 3000}},
+		},
+	}
+}
+
+func Get_utils_container_defaults() []Container {
+	return []Container{
+		{
+			Enabled: true,
+			Image:   "ghcr.io/dyrector-io/dyrectorio/web/kratos:latest",
+			Name:    "kratos-migrate",
+			Depends: []string{"- kratos-postgres"},
+			EnvVars: []EnvVar{
+				{"SQA_OPT_OUT", "true"},
+				{"DSN", "postgres://kratos:IPNuH10gAXT7bg8g@kratos-postgres:5432/kratos?sslmode=disable&max_conns=20&max_idle_conns=4"},
+			},
+			Volumes: []Volume{{"kratos-config", "/etc/config/kratos", true}},
+			Command: "-c /etc/config/kratos/kratos.yaml migrate sql -e --yes",
+		},
+		{
+			Enabled: true,
+			Image:   "ghcr.io/dyrector-io/dyrectorio/web/kratos:latest",
+			Name:    "kratos",
+			Depends: []string{"kratos-migrate:\n        condition: service_completed_successfully"},
+			EnvVars: []EnvVar{
+				{"SQA_OPT_OUT", "true"},
+				{"DSN", "postgres://kratos:IPNuH10gAXT7bg8g@kratos-postgres:5432/kratos?sslmode=disable&max_conns=20&max_idle_conns=4"},
+				{"KRATOS_ADMIN_URL", "http://kratos:9434"},
+				{"KRATOS_URL", "http://kratos:4433"},
+				{"AUTH_URL", "http://crux-ui:3000/auth"},
+				{"CRUX_UI_URL", "http://crux-ui:3000"},
+				{"DEV", "false"},
+				{"LOG_LEVEL", "info"},
+				{"LOG_LEAK_SENSITIVE_VALUES", "true"},
+				{"SECRETS_COOKIE", "AvGMteMxePpJ4ZVM"},
+				{"COURIER_SMTP_CONNECTION_URI", "smtps://${SMTP_USER}:${SMTP_PASSWORD}@${SMTP_URL}"},
+			},
+			Ports: []Port{
+				{9433, 4433},
+				{9434, 9434},
+			},
+		},
+		{
+			Enabled: true,
+			Image:   "docker.io/library/postgres:13.3-alpine3.14",
+			Name:    "crux-postgres",
+			EnvVars: []EnvVar{
+				{"POSTGRES_PASSWORD", "RhdAVuEnozSR4FLS"},
+				{"POSTGRES_USER", "crux"},
+				{"POSTGRES_DB", "crux"},
+			},
+			Ports:   []Port{{5432, 5432}},
+			Volumes: []Volume{{"crux-db", "/var/lib/postgresql/data", false}},
+		},
+		{
+			Enabled: true,
+			Image:   "docker.io/library/postgres:13.3-alpine3.14",
+			Name:    "kratos-postgres",
+			EnvVars: []EnvVar{
+				{"POSTGRES_PASSWORD", "IPNuH10gAXT7bg8g"},
+				{"POSTGRES_USER", "kratos"},
+				{"POSTGRES_DB", "kratos"},
+			},
+			Ports:   []Port{{5433, 5432}},
+			Volumes: []Volume{{"kratos-db", "/var/lib/postgresql/data", false}},
+		},
+	}
+}
+
+func Get_container_defaults(services []ServiceName) (error, []Container) {
+	var containers []Container
+
+	for _, i := range services {
+		switch i {
+		case CruxUI:
+			containers = append(containers, Get_cruxui_container_defaults()...)
+		case Crux:
+			containers = append(containers, Get_crux_container_defaults()...)
+		case Utils:
+			containers = append(containers, Get_utils_container_defaults()...)
+		default:
+			return errors.New("invalid service name"), []Container{}
+		}
+	}
+
+	return nil, containers
+}
+
+func Gen_container(services []ServiceName) (error, string) {
+	ReadConfig()
+	container_template := `version: "3.9"
```

Comment:
We can outsource these types of templates into files, IMHO. As we made in node script generation.

Reply:
Indeed, then embed it so the binary can be standalone.